rfcs/0001-tide-roadmap.md

```diff
@@ -0,0 +1,120 @@
+
+The current discussion is centered on providing a simple in memory session storage with Cookies as the default in the framework, and provide external crates to hook into external data stores.
+
+This change is currently blocked by the tide core changes.
+
+### Goals
+
+* [ ] Stable API.
+* [ ] In memory session backend
+
+## Authentication
+[Authentication]: #authentication
+
+Currently rust web frameworks have disparate ways of authentication. One of the goals of the Async ecosystem WG is to provide a common set of crates which can be used to build higher level abstractions. In terms of Tide, there currently isn't an authentication story present. @tomhoule has built cookie based authentication into the session management middleware as a POC. Therefore the goals for this sprint would be to sketch out the design of the authentication API for tide.
+
+### Goals
+
+* [ ] Design authentication API in Tide.
+
+[issues]: https://github.com/rustasync/tide/issues/
+[context-pr]: https://github.com/rustasync/tide/pull/156
+[issues-9]: https://github.com/rustasync/tide/issues/9
+[issues-99]: https://github.com/rustasync/tide/issues/99
+[issues-63]: https://github.com/rustasync/tide/issues/63
+[issues-24]: https://github.com/rustasync/tide/issues/24
+
+[sprint-goals]: https://github.com/rustasync/team/issues/96
+[goals-outline]: https://github.com/rustasync/team/issues/96#issuecomment-471552499
+[session-project]: https://github.com/tomhoule/tide-cookie-session
+
+# Websites
+[websites]: #websites
+
+There are two primary websites for keeping tack of async await progress and web development progress:
+
+* [Areweasyncyet][areweasyncyet]
+* [Arewewebyet][Arewewebyet]
+
+There isn't a lot that has been discussed around these two as they are in maintenance mode and work will mostly be around updating the content as and when appropriate.
+
+[areweasyncyet]: https://areweasyncyet.rs/
+[arewewebyet]: http://www.arewewebyet.org/
+
+# Async book
+[async-book]: #async-book
+
+The current [Async book][async-book] is a great start at explaining the details of why `async/await` is important, but is light on the details. As mentioned in [this issue][lucio-issue], there are discussions on improving the contents and providing a starting point for people wanting to learn about how the feature works. There are also plans for adding sections for helping maintainers of existing libraries to migrate to std futures.
+
+The book can also provide guides for building different kinds of software using `async/await` as well as provide guides for people wanting to help out with the documentation and migration effort, akin to [Tokio's doc-blitz][doc-blitz].
+
+
+### Goals
+
+Echoing the theme of the sprint, the goals for the book will be to provide a good foundation to build upon.
+
+TODO: Define specific goals?
```

Comment:
I think specific goals for 2019 is to provide a somewhat complete first draft of an async book, hopefully align it with the stable release of async/await?

Reply:
I wrote the goals section with the first 6 week sprint in mind. I can clarify that at the beginning of the document to make it clear. But now that I think about it, maybe the sprint goals should be in a separate section?

That said, what do you think the goals for the sprint should be?

Reply:
Yeah, I think it makes sense here to state the high-level goals of 2019. Then use another issue to express the 6-week sprint goals.

I think 6-week sprint as of now should be to outline the chapters, move the async-book over, start writing them. Don't know how much else we can expect though.